File: src/main/java/de/asmax/simpleperms/commands/override/DeopCommand.java
package de.***REMOVED***.simpleperms.commands.override;

import de.***REMOVED***.simpleperms.Main;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerCommandPreprocessEvent;

public class DeopCommand implements Listener {

    @EventHandler
    public void handleCommand(PlayerCommandPreprocessEvent event) {
        String message = event.getMessage();
        Player player = event.getPlayer();

        if(message.equalsIgnoreCase("/deop")) {
            event.setCancelled(true);
            Main.getInstance().getPermissionManager().removePermission(player, "*");
            Main.getInstance().getPermissionManager().savePlayerPermissions(player, "*");
            player.sendMessage(Main.getInstance().getPrefix() + "§4You have been successfully deoped!");
        }
    }
}

File: src/main/java/de/asmax/simpleperms/commands/override/DeopCommand.kt
package de.***REMOVED***.simpleperms.commands.override

import de.***REMOVED***.simpleperms.Main
import de.***REMOVED***.simpleperms.permissions.PermissionManager
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerCommandPreprocessEvent

class DeopCommand : Listener {
    @EventHandler
    fun handleCommand(event: PlayerCommandPreprocessEvent) {
        val message = event.message
        val player = event.player
        if (message.equals("/deop", ignoreCase = true)) {
            event.isCancelled = true
            PermissionManager.removePermission(player, "*")
            PermissionManager.savePlayerPermissions(player, "*")
            player.sendMessage("${Main.prefix}§4You have been successfully deoped!")
        }
    }
}